profiler/profiler.go

```diff
@@ -134,6 +138,9 @@ func newProfiler(opts ...Option) (*profiler, error) {
 	for _, opt := range opts {
 		opt(cfg)
 	}
+	if len(cfg.customProfilerLabels) > customProfileLabelLimit {
+		cfg.customProfilerLabels = cfg.customProfilerLabels[:customProfileLabelLimit]
+	}
```

Comment:
What's the reason for enforcing this in the client rather than the backend?

Reply:
I checked in with Alex on this. Full details in the linked JIRA. But to summarize, there aren't any hard limits on the backend for this, but we can change that quickly if needed. But he has suggested we start with a low limit and track usage to see if our users want more.

Reply:
SGTM!